### DISCUSSION TITLE
Viable for games ?

### DISCUSSION THREAD

**Discussion by Unreal852** [dotnet/Silk.NET#438] [Q&A] [answered]
Hello, I would like to know if Silk is viable for 3D game development ? How does it compare against OpenTK ?

**Comment by Perksey** [comment, marked as answer]
> Hello, I would like to know if Silk is viable for 3D game development ?

Yes! We have a number of people building games using Silk.NET; the largest of which is probably [Project Hedra](https://projecthedra.com) which has been using Silk.NET since the earliest days of our development.

> How does it compare against OpenTK?

I hate answering these questions because it's a case of "use which library works for you", but:

- OpenTK currently doesn't have a .NET Standard build that works on mobile. Silk.NET, on the other hand, is proven to work on Android and an iOS simulator.
- OpenTK doesn't have a consinstent release schedule, whereas you can count on us having an update every month - usually on the first Friday.
- Silk.NET is supported by the .NET Foundation, which homes a lot of prestigious projects and guarantees that we won't die tomorrow.
- The Silk.NET team has spent countless hours hand-optimizing the JIT assembly generated by our bindings and utility applications, so you can count on Silk.NET being nice and performant.
- Silk.NET, instead of having a static class, has a concept of an "API object". This is an inconvenience for some, but is done to maintain correctness with using lots of different contexts such that our library is scalable; and once you're used to it you'll really like the benefits it presents such as defining extension methods.
- We do what we can to make our bindings easy to work with, such as adding overloads for refs, spans, and more; and are constantly looking to improve this.
- Silk.NET is always up to date - we regenerate our bindings every month to keep up with the changes in the industry.

**Reply by BuslikDrev** [reply]
> * The Silk.NET team has spent countless hours hand-optimizing the JIT assembly generated by our bindings and utility applications, so you can count on Silk.NET being nice and performant.

Hello, did you compare performance with known game engines (unity, cryengine, unreal)? Is there an example of a render, for example, 10000 cubes that move?

**Reply by Perksey** [reply]
No.

**Comment by Unreal852** [comment]
Hi, thanks you, this is the answer i was looking for. I'll go with Silk.
